src/Parsers/ParserDropQuery.cpp

```diff
@@ -79,7 +80,7 @@ bool parseDropQuery(IParser::Pos & pos, ASTPtr & node, Expected & expected, bool
                 return false;
         }
 
-        if (s_no_delay.ignore(pos, expected))
+        if (s_no_delay.ignore(pos, expected) || s_sync.ignore(pos, expected))
             no_delay = true;
```

Comment:
Just out of curiosity why `SYNC` == `NO DELAY`, i.e. why sync drop calls `no delay`?

Reply:
Because I've previously added `NO DELAY` mode to ignore `database_atomic_delay_before_drop_table_sec` setting (8 minutes by default) and remove table data immediately when it's refcount become 0, but it was not helpful, because background removal was asynchronous anyway, so I've added synchronous mode.
`SYNC` mode obviously should imply [previous version of] `NO DELAY` mode, because if you want to wait for table to be actually dropped, 8 minutes timeout is excess. But it's hard to imagine a use case, when you want to ignore drop delay, but don't want to use synchronous mode, so `SYNC` and `NO DELAY` are synonyms now.

Reply:
Came to this https://github.com/ClickHouse/ClickHouse/blob/cc9dafcb309f750cb427a639a17d324f662ab17b/src/Interpreters/DatabaseCatalog.cpp#L701 and everythin become clear
Although this is pretty non obvious naming, what I was thought about NO DELAY is that do not delay DROP and just return async, while it was do not delay the DROP in the drop queue